### ISSUE TITLE
Standardize arrow functions

### ISSUE THREAD

**Issue by NicoGorr** [standard/standard#1144]
Could you please add some rules to arrow functions so the code will be more consistent?

I suggest:
`arrow-body-style` set to `as-needed`
`arrow-parens` set to `always`
`implicit-arrow-linebreak` set to `beside`

Thank you.

**Comment by LinusU** [comment]
I think this would be great as well, my own code is a total mix between `(err) => ...` and `err => ...`, sometimes even in the same project 😆 

I'm okay with both, but I think that always having the parenthesis is easier...

**Comment by NicoGorr** [comment]
Yes, it is. Using parens is make the code more consistent.

**Comment by LinusU** [comment]
Looking at the current eco-system impact of enabling `arrow-parens`:

```text
"arrow-parens": ["error", "always"]

# tests 271
# pass  246
# fail  25
# violations 696
```

```text
"arrow-parens": ["error", "as-needed"]

# tests 271
# pass  244
# fail  27
# violations 616
```

Hmm, this one is very very close.

I'm looking at being able to fetch the full list of packages depending on `standard` directly from Npm, instead of using our manually sampled list. Perhaps that can shed some more light on it...

**Comment by mightyiam** [comment]
```js
'arrow-body-style': ['error', 'as-needed', { requireReturnForObjectLiteral: false }]
```

Because I've been using the `() => ({})` pattern for a long while and I am used to reading it. I like it that way.

`'arrow-parens': 'always'`. I used to prefer `'as-needed'` but since writing so much strict TypeScript, it has been impossible to omit the parens. So yea. `'always'`.

I could live with `'implicit-arrow-linebreak': ['error', 'beside']`. I could not live with `'below'`.

**Comment by LinusU** [comment]
My vote also goes to `'arrow-parens': 'always'`

**Comment by razor-x** [comment]
I think `'implicit-arrow-linebreak': ['error', 'beside']` is important to add IMO. This fixes big standardization conflicts on my projects. When code uses the below linebrake style it essentially adds an extra level of indentation for every higher order function when compared with functions that use the below style.

(Standardizing on the `below` option instead would also normalize the indentation, but I am personally biased towards `beside`, and I get the sense that `beside` is more commonly used in projects.)

Can you confirm if this and any other options will be set for standard 15 since this was added to the milestone? 🙏♥️

**Comment by mightyiam** [comment]
If the body-less arrow function is short enough to be a one-liner, it seems to me that it should be a one-liner:

```js
const hasFoo = (o) => o.foo !== undefined
```

And there's no point in breaking like so:

```js
const hasFoo = (o) =>
  o.foo !== undefined
```

But for a longer body-less arrow function, which we would break at least once, I would prefer `below`:

```js
const isParticularState = (state) =>
  state.foo === 'something' &&
  state.bar !== NAZ &&
  state.baz &&
  state.n < 7
```

And I don't like how `beside` looks like in that case:

```js
const isParticularState = (state) => state.foo === 'something' &&
  state.bar !== NAZ &&
  state.baz &&
  state.n < 7
```

What I just described does not seem to be supported by [the rule](https://eslint.org/docs/rules/implicit-arrow-linebreak). I imagine that contributing such support would be quite a simple contribution.

**Comment by razor-x** [comment]
I agree. Note that you can still get close to you last example with beside using

```js
const isParticularState = (state) => (
  state.foo === 'something' &&
  state.bar !== NAZ &&
  state.baz &&
  state.n < 7
)
```

**Comment by mightyiam** [comment]
@razor-x that might even be more clear... Would anyone else like to provide input?

**Comment by jimmywarting** [comment]
i kind of like the shortest possible solution, requires fewer bytes/keystrokes 🙃 

```js
/** @param {number} ms */
const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))

/** @param {number} ms */
const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms)) // thinks this needs more reading...
```

- `arrow-body-style=as-needed` 👍 
- `arrow-parens` 🙁 would it be possible to have something like "consistent" and just pick the one preferences that occur the most in a project? if there is 3 similar `x = arg1 => fn(arg1)` and only one `x = (arg1) => fn(arg1)` then use `as-needed` and `always` if it's the other way around?

b/c it's such a close draw between using it and not using arrow-parens then i think we should leave it as is

**Comment by voxpelli** [comment]
@jimmywarting I think this does the trick? Its what I would want:

```json
"arrow-parens": ["error, "as-needed", { "requireForBlockBody": true }]
```
